src/DETHRACE/common/input.c

```diff
@@ -403,13 +403,15 @@ int AddRollingLetter(char pChar, int pX, int pY, tRolling_type rolling_type) {
     let->current_offset = (gCurrent_graf_data->save_slot_letter_height * let->number_of_letters);
     for (i = 0; i < let->number_of_letters; i++) {
         if (rolling_type == eRT_numeric) {
-            let->letters[i] = pChar;
+            /* The (tU8) cast makes sure extended ASCII is positive. */
+            let->letters[i] = (tU8)pChar;
```

Comment:
This is done in the German Carmageddon DOS executable.

This makes sure the extended ASCII codes remain positive.
Before, these became negative here.

Not doing this causes an out-of-bounds access when reading the texture of an extended ASCII character.